score/score.go

```diff
@@ -0,0 +1,141 @@
+package score
+
```

Comment:
Most of these functions require calculating combinations of cards. These combinations are recalculated over and over - can we memoize to make this more optimal?